### ISSUE TITLE
Weighted glmer model unexpected dependency on data frame

### ISSUE THREAD

**Issue by mdg583** [lme4/lme4#589]
After building a weighted binomial model using the glmer function, calls to `confint` or `profile` fail after any changes to the data frame in the calling environment. This is also posted [here](https://stackoverflow.com/questions/63669058/avoid-failure-of-confint-mermod-on-weighted-models-in-lme4-when-data-object-modi). A simple example is:

```
library(lme4)

set.seed(1)
n <- 1000
df <- data.frame(
  y=rbinom(n,1,.5),
  w=runif(n,0,1)*.1+.95,
  g=as.integer(round(runif(n,0,4)))
)
m <- glmer(cbind(y,1-y)~(1|g),data=df,weights=w,family=binomial())
confint(m)
df$w <- df$w*2
confint(m)
```
The 2nd call to `confint` gives:
```
Computing profile confidence intervals ...
Error in profile.merMod(object, which = parm, signames = oldNames, ...) : 
  Profiling over both the residual variance and
fixed effects is not numerically consistent with
profiling over the fixed effects only
```

This dependency on the data frame in the calling environment is unexpected.

R version 3.6.3 (2020-02-29), x86_64-redhat-linux-gnu, lme4_1.1-21

**Comment by bbolker** [comment]
Internally, constructing the modified deviance function for profiling `glmer` fits relies on `update()` to reconstruct the original deviance function (see [here](https://github.com/lme4/lme4/blob/master/R/profile.R#L522)), which in turn relies on re-evaluating the model in the original environment. In principle it should be possible to recreate the deviance function from internal structures rather than going back to the original environment ...

How important is this/how hard is it to work around ... ?

**Comment by bbolker** [comment]
This shows a potential solution for reconstructing a new deviance function from a `glmer` object ... still needs to be incorporated and tested.

```
library(lme4)
gm1 <- glmer(cbind(incidence, size - incidence) ~ period + (1 | herd),
                   data = cbpp, family = binomial, nAGQ=1)

getdevfun <- function(object) {
    ## FIXME: make sure this works for different nAGQ etc etc.
    reTrms <- getME(object,c("Zt","theta","Lambdat","Lind","flist","cnms"))
    d1 <- mkGlmerDevfun(object@frame, getME(object,"X"),
                        reTrms=getME(object,c("Zt","theta","Lambdat","Lind")),
                        family(object))
    nAGQ <- object@devcomp$dims[["nAGQ"]]
    d2 <- updateGlmerDevfun(d1, reTrms, nAGQ=nAGQ)
    d2
}

dd <- getdevfun(gm1)
pp <- unlist(getME(gm1,c("theta","beta")))

## diff somewhat but maybe not too much ...
all.equal(-2*c(logLik(gm1)),dd(pp),tolerance=0)
```

**Comment by mmaechler** [comment]
Even if there's a work around,  this issue should eventually be solved.  It's not acceptable that the glmer() object itself is insufficient to compute derived quantities/properties.    
An easier alternative may be to have  `confint()` explicitly require not only the fitted object but also the `data` -- but I don't like that alternative either.

**Comment by bbolker** [comment]
Should be resolved in 8ade709cd3 . Constructing the deviance function directly rather than via `update()` leads to *slight* numerical differences in results (digging out why is theoretically possible but beyond me at this point ...).  This has been noted in `NEWS.Rd`.
